electrum/gui/qt/transaction_dialog.py

```diff
@@ -245,6 +251,9 @@ def insert_tx_io(
                 tcf_shortid=tcf_shortid,
                 short_id=str(txin.short_id), addr=addr, value=txin_value,
             )
+            if isinstance(txin, PartialTxInput):
+                if txin.sighash != Sighash.ALL and self.wallet.adb.is_mine(addr):
```

Comment:
I think it would be better to move the checks to the shared lib, have that return a warning, and only add GUI-specific code that can show warnings.

In particular, have you seen `wallet.get_warning_for_risk_of_burning_coins_as_fees(tx)`?
That already exists (not used in qml though), we could maybe just rename and extend that with this new kind of warning.

Another argument for trying to have a generic warning logic instead of different kind of warning tests directly done in the GUI is that `wallet.get_warning_for_risk_of_burning_coins_as_fees` was used in both qt and kivy, but none of us noticed it was not added to qml: fewer things to do in the GUI, the better. :)

Reply:
I moved the check to the backend, and made a distinction between confirmation needed and reject. 

Not sure if this should stay within `PartialTransaction` or if `Wallet` is a better place for it..

I don't want to overload `wallet.get_warning_for_risk_of_burning_coins_as_fees` with this, as that func name is rather specific, but we can add a high-level 'do-all-checks' func which calls the specific ones.

Reply:
I'll make a TODO item for `wallet.get_warning_for_risk_of_burning_coins_as_fees` in qml

Reply:
Note that `wallet.get_warning_for_risk_of_burning_coins_as_fees` is closely related in the sense that it too is doing sighash stuff. Specifically, SIGHASH_ALL is not always safe either, depending on legacy/segwit/taproot, the number of inputs, etc, and that is what is being considered there.

Reply:
> Not sure if this should stay within `PartialTransaction` or if `Wallet` is a better place for it..

Moved `check_sighash` to `Abstract_Wallet`, seems more appropriate.